### PR TITLE
File watcher update

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Implement IVsFreeThreadedFileChangeEvents2 
This gives us more info about changes files, and allows us to remove a bunch of no longer used code.

### CODE REVIEW

File: Nodejs/Product/Nodejs/SharedProject/AssemblyReferenceNode.cs [hunk 1]
```diff
@@ -400,7 +400,7 @@ protected virtual void OnAssemblyReferenceChangedOnDisk(object sender, FileChang
             }
 
             // We only care about file deletes, so check for one before enumerating references.
-            if ((e.FileChangeFlag & _VSFILECHANGEFLAGS.VSFILECHG_Del) == 0)
+            if (e.FileChange != WatcherChangeTypes.Deleted)
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/SharedProject/AssemblyReferenceNode.cs]: >WatcherChangeTypes.Deleted [](start = 32, length = 26)

Is this a flags enum?  If so, do you still need a containment check?

File: Nodejs/Product/Nodejs/SharedProject/CommonProjectNode.cs [hunk 2]
```diff
@@ -728,17 +728,10 @@ protected override ReferenceContainerNode CreateReferenceContainerNode()
             return new CommonReferenceContainerNode(this);
         }
 
-        private void FolderChangedOnDisk(object sender, FolderChangedEventArgs e)
+        private void FileChangedOnDisk(object sender, FileChangedOnDiskEventArgs e)
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/SharedProject/CommonProjectNode.cs]: >FileChangedOnDisk [](start = 21, length = 17)

Ben had some problems recently where moving a folder didn't produce file-changed events for the files in the folder.  Does this API behave more reasonably?

File: Nodejs/Product/Nodejs/SharedProject/StructuresEnums.cs [hunk 3]
```diff
@@ -315,7 +317,26 @@ internal FileChangedOnDiskEventArgs(string fileName, uint id, _VSFILECHANGEFLAGS
...
             this.ItemID = id;
-            this.FileChangeFlag = flag;
+            this.FileChange = ConvertVSFILECHANGEFLAGS((uint)flag);
+        }
+
+        private static WatcherChangeTypes ConvertVSFILECHANGEFLAGS(uint flag)
+        {
+            if ((flag & (uint)(_VSFILECHANGEFLAGS.VSFILECHG_Size | _VSFILECHANGEFLAGS.VSFILECHG_Time | _VSFILECHANGEFLAGS.VSFILECHG_Attr)) != 0)
+            {
+                return WatcherChangeTypes.Changed;
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/SharedProject/StructuresEnums.cs]: >return [](start = 16, length = 6)

Other bits in `flag` can be ignored?

**Comment by paulvanbrenk** [comment]
@uniqueiniquity it's a lot less resource consuming in VS.

**Pull request merged**